### PR TITLE
feat: add table component

### LINKED ISSUE #23: テーブルコンポーネントの作成

### PR DESCRIPTION
**Pull request by enzinia**
Added very basic Table component.
Support JSX.

Close #23

### CODE REVIEW

**Comment by 9renpoto** [comment]
https://ant.design/components/table/ How about this? :)

File: packages/components/src/Table/Table.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  border-spacing: 0;
+  border-collapse: separate;
+  border-radius: 2px 2px 0 0;
+`;
+
+const StyledTableHead = styled.thead`
+  & > tr > th {
+    position: relative;
+    padding: 16px;
+    font-weight: 500;
```
Comment (sottar) [inline packages/components/src/Table/Table.tsx:41]: I think we don't need this line.

File: packages/components/src/Table/Table.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    text-align: left;
+    overflow-wrap: break-word;
+    background: #fafafa;
+    border-bottom: 1px solid #f0f0f0;
+    transition: background 0.3s ease;
+  }
+`;
+
+const StyledTableBody = styled.tbody`
+  & > tr {
```
Comment (sottar) [inline packages/components/src/Table/Table.tsx:52]: Looks you don't need `&`

File: packages/components/src/Table/Table.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    & > td {
```
Comment (sottar) [inline packages/components/src/Table/Table.tsx:53]: [Ditto] Looks you don't need `&`

File: packages/components/src/Table/Table.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import styled from 'styled-components';
+
+type ColumnsType = { [index: string]: string }[];
```
Comment (sottar) [inline packages/components/src/Table/Table.tsx:4]: You should make this type more concretely.
like 
```
type ColumnsType = { key: string; title: string }[];

```

File: packages/components/src/Table/Table.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+const StyledTable = styled.table`
+  width: 100%;
+  text-align: left;
+  border-spacing: 0;
+  border-collapse: separate;
+  border-radius: 2px 2px 0 0;
+`;
+
+const StyledTableHead = styled.thead`
+  & > tr > th {
```
Comment (sottar) [inline packages/components/src/Table/Table.tsx:38]: Looks you don't need `&`

File: packages/components/src/Table/Table.tsx [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+type ColumnsType = { [index: string]: string }[];
+type DataType = { [index: string]: string | number | JSX.Element | JSX.Element[] }[];
+
+export type TableProps = {
+  columns: ColumnsType;
+  data: DataType;
+  className?: string;
+};
+
+const Wrapper = styled.div`
```
Comment (sottar) [inline packages/components/src/Table/Table.tsx:13]: Do you really need this Wrapper tag?
I feel we don't need it.

**Comment by sottar** [comment]
You also need to export Table component on `/packages/components/src/index.ts`

**Comment by enzinia** [comment]
@sottar Sorry, not noticed review notifications.
Fixed!

File: packages/components/src/Table/Table.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import styled from 'styled-components';
+
+type ColumnsType = { key: string; title: string }[];
+type DataType = { [index: string]: string | number | JSX.Element | JSX.Element[] }[];
```
Comment (sottar) [inline packages/components/src/Table/Table.tsx:5]: [nits] You can use `React.ReactNode` instead of `string | number | JSX.Element | JSX.Element[]`

**Review by sottar — APPROVED** [review]

File: packages/components/src/Table/Table.tsx (hunk 7 above)
Comment (enzinia) [reply packages/components/src/Table/Table.tsx:5]: Thanks, good to know!

**Comment by sottar** [comment]
@enzinia could you solve the conflict?

**Comment by enzinia** [comment]
Resovled!

**Review by sottar — APPROVED** [review]
Thanks!

**Pull request merged**